### PR TITLE
Charset detection fixes

### PR DESCRIPTION
**Pull request by phw**
# Summary

* This is a…
  * [x] Bug fix
  * [ ] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

This fixes a couple of issues of charset-normalizer is not installed:

- the utf-32-le BOM was in wrong order
- longer BOMs need to be checked first
- added BOM for utf-8-sig
- skip test for windows-1251 detection (only available with charset-normalizer or chardet)

### CODE REVIEW

File: test/test_utils.py [hunk 1]
```diff
@@ -960,7 +961,9 @@ def test_detect_file_encoding_bom(self):
                 f = NamedTemporaryFile(delete=False)
                 f.write(bom)
                 f.close()
-                self.assertEqual(expected_encoding, detect_file_encoding(f.name))
+                encoding = detect_file_encoding(f.name)
+                self.assertEqual(expected_encoding, encoding,
+                                 f'BOM {bom!r} dected as {encoding}, expected {expected_encoding}')
```
Comment (zas) [inline test/test_utils.py:966]:
```suggestion
                                 f'BOM {bom!r} detected as {encoding}, expected {expected_encoding}')
```

**Review by zas — CHANGES_REQUESTED** [review]

File: test/test_utils.py [hunk 2]
```diff
@@ -949,7 +950,7 @@ def test_detect_file_encoding_bom(self):
         boms = {
             b'\xff\xfe': 'utf-16-le',
             b'\xfe\xff': 'utf-16-be',
-            b'\00\00\xff\xfe': 'utf-32-le',
+            b'\xff\xfe\00\00': 'utf-32-le',
```
Comment (zas) [inline test/test_utils.py:953]: Good catch ;=)

**Review by zas — CHANGES_REQUESTED** [review]

File: test/test_utils.py (hunk 1 above)
Comment (phw) [reply test/test_utils.py]: Fixed

File: test/test_utils.py (hunk 2 above)
Comment (phw) [reply test/test_utils.py]: Fixed, thanks

**Comment by phw** [comment]
Minor addition to enable the multiline edit for the syncedlyrics tag: https://github.com/metabrainz/picard/commit/647fc3bd32f607d5868dda5614504cd207e9f8fa

Maybe we can improve this further in the future. Editing comments or lyrcis is still pretty awkward in Picard. And for synced lyrics we could even have a dedicated editor.

**Review by zas — APPROVED** [review]
LGTM

**Comment by zas** [comment]
@Sophist-UK can you re-review this PR?

**Review by Sophist-UK — APPROVED** [review]
Apologies for missing this. LGTM

**Pull request merged**